src/validator/YieldValidator.cpp

```diff
@@ -44,39 +51,21 @@ Status YieldValidator::validateImpl() {
         return Status::SemanticError("Only one variable allowed to use.");
     }
 
-    if (hasAggFun_) {
-        NG_RETURN_IF_ERROR(checkAggFunAndBuildGroupItems(yield->yield()));
-    }
-
-    if (exprProps_.inputProps().empty() && exprProps_.varProps().empty() && inputVarName_.empty()) {
+    if (!groupByValidator_ && exprProps_.inputProps().empty()
+        && exprProps_.varProps().empty() && inputVarName_.empty()) {
         // generate constant expression result into querycontext
         genConstantExprValues();
     }
 
     if (!exprProps_.varProps().empty() && !userDefinedVarNameList_.empty()) {
+        // TODO: Multiple userDefinedVars not support yet, may be supported later
+        DCHECK_EQ(userDefinedVarNameList_.size(), 1);
```

Comment:
here return error directly,  not use DCHECK.

Reply:
Modified~